### PR TITLE
fix(embed): fix multiple embeds (#1078)

### LINKED ISSUE #1078: Embed with :type=code replacing entire page when other embeds are present
## Bug Report
(potentially related to #791 and #866 or #970)

#### Steps to reproduce
Include a file as a `.html` and then as a code block, i.e.

```md
[embed](embed.html ':include')

[source](embed.html ':include :type=code :fragment=demo')
```

[raw files](https://gist.github.com/randName/9b2675aa92487228b18c81ff8bb14e1a)

[rendered version](https://bl.ocks.org/randName/raw/9b2675aa92487228b18c81ff8bb14e1a/)

Tested in Chrome, Firefox and Edge

#### What is current behaviour
The code block overwrites the page content (after a short delay).

The order of embedding does not matter, and even if the linked files are different this will happen.

If either block is removed there is no issue, and it seems unrelated to the fragment nor highlight type.

#### What is the expected behaviour
Multiple embed blocks remain visible.

#### Other relevant information

- [x] Bug does still occur when all/other plugins are disabled?

- Your OS: Windows 10 x64
- Browser version: Chrome 80.0.3987.132
- Docsify version: latest (4.11.2)
- Docsify plugins: None

### PR DESCRIPTION
**Pull request by madhavarshney**
**Summary**

After many hours of debugging 😅 , I found that `marked` mutates the `tokens` array that we pass directly, breaking the logic in `prerenderEmbed` for multiple embeds. This PR fixes #1078.

[See this in `marked`](https://github.com/markedjs/marked/blob/97be813c267b3d60a1bcef6f0b7ffa6014abc230/lib/marked.js#L1126-L1142).

https://github.com/docsifyjs/docsify/blob/2048610aacd4e3c6a592f4247834a726c7ca33fb/src/core/render/index.js#L169-L180

Also, this _may_ `fix` #791, though I haven't tested that yet.

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Repo settings
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

### CODE REVIEW

File: src/core/render/embed.js [hunk 1]
```diff
@@ -146,9 +146,10 @@ export function prerenderEmbed({ compiler, raw = '', fetch }, done) {
         .concat(embedToken, tokens.slice(index + 1));
       moves.push({ start: index, length: embedToken.length - 1 });
     } else {
+      const cloned = tokens.concat();
       cached[raw] = tokens.concat();
-      tokens.links = cached[raw].links = links;
-      done(tokens);
+      tokens.links = cloned.links = cached[raw].links = links;
+      done(cloned);
```
Comment (madhavarshney) [inline src/core/render/embed.js:152]: IMO all this cloning here is a bit odd...

**Comment by madhavarshney** [comment]
I'm currently a bit busy, so if anyone would like to add some tests, then go ahead! Otherwise, I'll try to do it soon (hopefully near the end of this week).

**Comment by anikethsaha** [comment]
Thanks for the PR. 

@madhavarshney take your time , no hurry at all. You can also convert this to draft if its ok by you in order to let others know that this is still WIP

**Review by anikethsaha — APPROVED** [review]
Need tests for this rest its ok.

**Review by Koooooo-7 — APPROVED** [review]

**Review by sy-records — CHANGES_REQUESTED** [review]
Waiting to add a test

**Comment by madhavarshney** [comment]
This has been fixed with the upgrade to `marked.js`.

**Pull request closed without merging**